torchmultimodal/modules/encoders/image_encoder.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Callable, Optional
+
+from torch import nn, Tensor
+from torchmultimodal.modules.layers.transformer import TransformerOutput
+
+
+class VisionTransformer(nn.Module):
+    """
+    General image transformer encoder with embeddings. Similar to ``VisionTransformer`` in torchvision,
+    but more composable. Can be constructed with any user-provided embeddings, encoder, and task head.
```

Comment:
another nit: I would take out the "more composable" part. Imo it's not necessary to explicitly say this

Reply:
Do you mean it's implied by the API that we don't have to explicitly say this? 
On the other hand, with docstring the wording should be descriptive while "more XXX" can feel a bit opinionated ? :)